src/com/untamedears/JukeAlert/util/Utility.java

```diff
@@ -68,19 +67,25 @@ public static void notifyGroup(Snitch snitch, String message) throws SQLExceptio
             iter.maxDistance(JukeAlert.getInstance().getConfigManager().getMaxAlertDistanceNs());
         }
         for (Player player : iter) {
-            RateLimiter.sendMessage(player, message);
+        	if (NameAPI.getGroupManager().hasAccess(snitch.getGroup(), player.getUniqueId(), PermissionType.getPermission("SNITCH_NOTIFICATIONS"))) {
+        		RateLimiter.sendMessage(player, message);
+        	}
         }
     }
-
-    public static boolean isOnSnitch(Snitch snitch, UUID accountId) {
-        Group faction = snitch.getGroup();
-        if (faction == null) return false;
-        return faction.isMember(accountId);
+    
+    public static boolean immuneToSnitch(Snitch snitch, UUID accountId) {
+    	Group group = snitch.getGroup();
+    	if(group == null) {
+    		return true;
+    	}
+    	//group object might be outdated so use name
+    	return NameAPI.getGroupManager().hasAccess(group.getName(), accountId, PermissionType.getPermission("SNITCH_IMMUNE"));
     }
     
     private static long failureReportDelay = 10000l;
     private static long lastNotifyPOSFailure = System.currentTimeMillis() - failureReportDelay;
     
+    /*
```

Comment:
Just delete these unused. No point leaving the dead code around, we can restore from history if needed.